source/codegen/metadata/nirfmxspecan_restricted/config.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+config = {
+    "api_version": "21.0.0",
+    "restricted_driver_service": True,
+    "c_header": "niRFmxSpecAn.h",
+    "c_function_prefix": "RFmxSpecAn_",
+    "service_class_prefix": "NiRFmxSpecAnRestricted",
+    "java_package": "com.ni.grpc.nirfmxspecanrestricted",
+    "csharp_namespace": "NationalInstruments.Grpc.NiRFmxSpecAnRestricted",
+    "namespace_component": "nirfmxspecan_restricted",
+    'close_function': None,
+    "custom_types": [],
+    "type_to_grpc_type": {
+        "char[]": "string",
+        "float32": "float",
+        "float64": "double",
+        "int16": "int32",
+        "int32": "int32",
+        "int64": "int64",
+        "niRFmxInstrHandle": "nidevice_grpc.Session",
+        "int8": "int32",
+        "uInt16": "uint32",
+        "uInt32": "uint32",
+        "uInt64": "uint64",
+        "uInt8": "uint32",
+        "uInt8[]": "bytes",
+        "NIComplexSingle": "nidevice_grpc.NIComplexNumberF32",
+        "NIComplexDouble": "nidevice_grpc.NIComplexNumber",
+    },
+    "driver_name": "NI-RFMXSPECAN-RESTRICTED",
```

Comment:
I'm thinking that we should add 'code_readiness': 'Prototype', or maybe a new type of "restricted" so this doesn't get included by default in our client artifacts (proto and examples). @astarche , thoughts on this? I think then the clients (like InstrumentStudio) that do want to use this restricted service can enable it on the server through the config file.

Reply:
I tried with code_readiness: Prototype and my tests started failing. I debugged this and it is because my tests are using DeviceServer in tests/system/device_server.cpp and this class only supports NextRelease and up. Should I make DeviceServer work with Prototype (instead of NextRelease)?

Reply:
I like the idea to distinguish internal APIs with `code_readiness`. I think our current set of readiness states doesn't represent this well .

I think we should **avoid** having the "released" state of internal APIs be "prototype". I think it would also be helpful to have the "pre-release" state of internal apis be **different** from the pre-release state of (future) public APIs. But I don't think that's an absolute requirement.

I think this would be good:

```
InternalNextRelease, // Dev state for internal APIs
NextRelease,
Internal, // release state for internal APIs
Release
```

I like "internal", "restricted" or "unstable" as the qualifier for internal APIs. Any of those are fine.

I'd be OK with using `Prototype` instead of `NextReleaseInternal` for the dev state. But `Prototype` has a different meaning in our other apps, so I don't like that as much.

We borrowed the terms for readiness from ASW, which is not a perfect fit for this problem. If either of y'all want to consider a more significant rewording, here's an idea:

```
Development, // Dev state for Internal APIs and APIs that are not approaching release
Unstable, // Release state for Internal APIs
Hardening, // Dev state for Public APIs as they approach release
Release
```

Also note that `NextRelease` and `Internal` APIs may have the same behavior. That would make sense to a client. Using a pre-release driver API should be more-or-less similar to using an internal/unstable API that we don't intend to harden.

@madrasm Yes. Whatever we choose as the "dev state of internal APIs" should be the lowest readiness enabled by `device_server`. We've never used the "readiness" states that are "lower" than what `device_server` enables. Those lower "readiness" states are a bit dubious. The reason we would use them is to stage a change that has the potential to destabilize the system test suite (but that we still want to test with integration tests). That would not be a driver readiness, but possible the readiness of some internal feature. We don't need to worry about that too much, but whatever change we make we can document `readiness_enabled_by_device_server - 1 => a_dev_feature_that_we_want_to_disable_in_auto_tests`. In ASW, that readiness is `prototype`, but if it's `incomplete` or `tbd` that's OK too.

Reply:
Fixed in commit [b63bc2e](https://github.com/ni/grpc-device/pull/689/commits/b63bc2e2ccc3599d68327e9a5f4c8fcc41e95dc0). Went with 'Restricted' in the new code readiness fields to match with the nomenclature in the new files, etc.